src/main/java/com/websafe/ccmsbe/service/PrivacyRegulationService.java

```
@@ -0,0 +1,148 @@
+            // Handle other exceptions
+            throw new PrivacyRegulationCreationException("An error occurred while creating the privacy regulation");
+        }
+    }
+    public List<PrivacyRegulation> getAllPrivacyRegulations() {
+        try {
+            return privacyRegulationRepository.findAll();
+        } catch (Exception e) {
+            // Handle the exception
+            throw new PrivacyRegulationRetrievalException("An error occurred while retrieving all privacy regulations");
+        }
+    }
+//    public Optional<PrivacyRegulation> getPrivacyRegulationsFromWebsite(Long websiteId) {
+//        return privacyRegulationRepository.findById(websiteId);
+//    }
+    @Transactional
+    public String addPrivacyRegulationToWebsite(Long websiteId, Long regulationId) {
+        try {
+            Website website = websiteRepository.findById(websiteId).orElse(null);
+            PrivacyRegulation privacyRegulation = privacyRegulationRepository.findById(regulationId).orElse(null);
+
+            // Check for null entities
+            if (website == null || privacyRegulation == null) {
+                throw new PrivacyRegulationNotFoundException("Invalid website ID or regulation ID");
+            }
+
+            // Check for duplication
+            if (website.getPrivacyRegulations().contains(privacyRegulation)) {
+                throw new PrivacyRegulationNotFoundException("Regulation already added to the website");
+            }
+
+            website.getPrivacyRegulations().add(privacyRegulation);
+            websiteRepository.save(website);
+            return "Successfully added";
+        } catch (PrivacyRegulationNotFoundException e) {
+            // Handle the custom exception appropriately
+            return e.getMessage(); // Return the error message
+        } catch (Exception e) {
+            // Handle other exceptions if needed
+            return "An error occurred while adding the privacy regulation";
```

Let's throw an exception instead.